tempesta_fw/http.c

```diff
@@ -2133,8 +2133,21 @@ static int
 tfw_http_adjust_resp(TfwHttpResp *resp)
 {
 	TfwHttpReq *req = resp->req;
-	int r, conn_flg = req->flags & __TFW_HTTP_MSG_M_CONN_MASK;
 	TfwHttpMsg *hm = (TfwHttpMsg *)resp;
+	int r, conn_flg;
+
+	/*
+	 * Request violated backend rules, so backend closed connection to
+	 * Tempesta. Close client connection after forwarding the response.
+	 */
+	if ((resp->flags & TFW_HTTP_F_CONN_CLOSE) && (resp->status != 200)) {
```

Comment:
I believe not only 200 should be treated as "good" response, at least `304 Not Modified`, `204 No Content`, `206 Partial Content` and I think quite a few others should be treated w/o connection closing. Please check an RFC and/or a reference implementation (e.g. Nginx or Apache) about the cases when a client connection is closed. I assume it should be only for client error (4xx) codes. Please add a comment here about the standard statements.

Reply:
I forgot to mention that the scenario is pretty trivial, so I wonder why don't we have it in our functional tests? Please also write a functional test which checks all the status codes and connection resets according to the standard.